### PR TITLE
feat: add docker configuration

### PR DESCRIPTION
**Pull request by sileht**
Initial step to be able to deploy the engine as docker image

Fixes MRGFY-12

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RUN sed -e '/^-e ./d' /requirements.txt > /constraints.txt
+RUN python3 -m pip install --no-cache-dir -r /constraints.txt
+
+# Real install that can't be cached
+ADD . /app
+WORKDIR /app
+RUN python3 -m pip install --no-cache-dir -r ./requirements.txt
+
+
+FROM python:3.9-alpine as runner
```
Comment (jd) [inline Dockerfile:18]: can we use ubuntu?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -75,3 +75,9 @@ jobs:
       - run: |
           pip install tox
           tox -e docs
+
+  docker:
+    runs-on: ubuntu-20.04
```
Comment (jd) [inline .github/workflows/ci.yaml:80]: not latest?

**Review by jd — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (sileht) [reply .github/workflows/ci.yaml:80]: latest is randomly 18.04 or 20.04 currently (https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on)

File: .github/workflows/docker.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: Publish Docker image
+on:
+  release:
+    types: [published]
+jobs:
+  push_to_registry:
+    name: Push Docker image to GitHub Packages
+    runs-on: ubuntu-latest
```
Comment (jd) [inline .github/workflows/docker.yaml:8]: do we want latest or 20.04 then here?

Comment (jd) [reply .github/workflows/docker.yaml:8]: ping @sileht

**Review by jd — CHANGES_REQUESTED** [review]

**Review by jd — APPROVED** [review]

File: setup.cfg [hunk 4]
```diff
@@ -24,6 +24,7 @@ install_requires =
     datadog
     python-dotenv
     fastapi
+    aiofiles
```
Comment (jd) [inline setup.cfg:27]: why?
you missed the asgi file I guess

File: .gitignore [hunk 5]
```diff
@@ -5,6 +5,7 @@ AUTHORS
 cover
 .coverage
 dist
+.mypy_cache
```
Comment (jd) [inline .gitignore:8]: https://github.com/Mergifyio/mergify-engine/pull/2060

**Review by jd — CHANGES_REQUESTED** [review]

File: setup.cfg (hunk 4 above)
Comment (sileht) [reply setup.cfg:27]: That's used by the installer, that shouldn't be there

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+cd /app
+
+if [ "$MERGIFYENGINE_INTEGRATION_ID" ]; then
+  case $1 in
+      web) exec gunicorn -k uvicorn.workers.UvicornH11Worker --statsd-host localhost:8125 --log-level warning mergify_engine.asgi;;
```
Comment (jd) [inline entrypoint.sh:6]: this ain't gonna work

**Review by jd — CHANGES_REQUESTED** [review]

File: entrypoint.sh (hunk 6 above)
Comment (sileht) [reply entrypoint.sh:6]: why ?

Comment (jd) [reply entrypoint.sh:6]: I misread 😢

**Review by jd — APPROVED** [review]

**Pull request merged**